Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -50,40 +56,71 @@ public static class CustomEntitySearch
             {
                 return new BadRequestObjectResult($"{skillName} - Invalid request record array.");
             }
-
+            
             WebApiSkillResponse response = WebApiSkillHelpers.ProcessRequestRecords(skillName, requestRecords,
-                (inRecord, outRecord) => {
+                 (inRecord, outRecord) => {
+                    if (!inRecord.Data.ContainsKey("text") || inRecord.Data["text"] == null)
+                    {
+                        outRecord.Errors.Add(new WebApiErrorWarningContract { Message = "Cannot process record without the given key 'text' with a string value" });
+                        return outRecord;
+                    }
+                    if (!inRecord.Data.ContainsKey("words") && 
+                    (inRecord.Data.ContainsKey("synonyms") || inRecord.Data.ContainsKey("exactMatches") || inRecord.Data.ContainsKey("fuzzyMatchOffset")))
+                    {
+                        outRecord.Errors.Add(new WebApiErrorWarningContract {
+                            Message = "Cannot process record without the given key 'words' in the dictionary"});
+                        return outRecord;
+                    }
                     string text = inRecord.Data["text"] as string;
-                    IList<string> words = ((JArray)inRecord.Data["words"]).ToObject<List<string>>();
-                    if (words == null)
+                    IList<string> words = inRecord.GetOrCreateList<List<string>>("words");
+                    Dictionary<string, string[]> synonyms = inRecord.GetOrCreateDictionary<Dictionary<string, string[]>>("synonyms");
+                    IList<string> exactMatches = inRecord.GetOrCreateList<List<string>>("exactMatches");
+                    int offset = (inRecord.Data.ContainsKey("fuzzyMatchOffset")) ? Math.Max(0, Convert.ToInt32(inRecord.Data["fuzzyMatchOffset"])) : 0;
+                    bool caseSensitive = (inRecord.Data.ContainsKey("caseSensitive")) ? (bool)inRecord.Data.ContainsKey("caseSensitive") : false;
+                    if (words.Count == 0 || (words.Count(word => !String.IsNullOrEmpty(word)) == 0))
                     {
-                        words = new WordLinker(executionContext.FunctionAppDirectory).Words;
+                        outRecord.Warnings.Add(new WebApiErrorWarningContract {
+                            Message = "Used predefined key words from customLookupSkill configuration file " +
+                            "since no 'words' parameter was supplied in web request" });
+                        WordLinker userInput = WordLinker.WordLink(executionContext.FunctionAppDirectory);
+                        words = userInput.Words;
```

Comment:
can you try/catch this and provide a specific error message like "could not parse predefined words.json" instead of some potentially ambiguous "failed to deserialize" error